src/App.php

```diff
@@ -1011,19 +1010,28 @@ public function runFrontend()
 	{
 		// Missing DB connection: ERROR
 		if ($this->getMode()->has(App\Mode::LOCALCONFIGPRESENT) && !$this->getMode()->has(App\Mode::DBAVAILABLE)) {
-			Core\System::httpExit(500, ['title' => 'Error 500 - Internal Server Error', 'description' => 'Apologies but the website is unavailable at the moment.']);
+			echo Module\Special\HTTPException::rawContent(
```

Comment:
The problem I currently have for this approach is that this special module uses templating. At the moment we don't seem to use config values in the current template engine but it may happen. I'm afraid of circular dependency between the error module and the templating engine in case the App isn't properly initialized.

If we ignore that, the other question I had is if I could have a try/catch block inside another. Here's my ideal structure that I don't know is feasible:
```php
class App {
    ...
    public function runFrontend() {
        try {
            // Missing DB connection: ERROR
            if ($this->getMode()->has(App\Mode::LOCALCONFIGPRESENT) && !$this->getMode()->has(App\Mode::DBAVAILABLE)) {
                throw new HTTPException\InternalServerErrorException('Apologies but the website is unavailable at the moment.');
            }
           
            ...
           
            try {
                // Processing module content
            } catch(HTTPException $e) {
                $content = Module\Special\HTTPException::content($e);
            }
           
            ...
           
            require_once $template;
        } catch(HTTPException $e) {
            $content = Module\Special\HTTPException::rawContent($e);
            System::httpExit($e->getCode(), $e->httpdesc, $content);
        }
    }
    ...
}
```

Reply:
hm .. I see a more "radical" approach for the future:

index.php:
```php
$dependencyManager = new DependencyManager(Dice $diceDILibrary);
$router = new Router($dependencyManager);

$module = $router->getModule();

$module->run();
```

Router:
```php
class Router {
     /**
      * @return IModule
      */
    public function getModule()
    { 
        try {
           $moduleName = $this->getModuleBecauseOfRoutes();
       
           // the dependency manager creates the IModule for us with all dependencies (including App, DBA, ...)
           return $this->dependencyManager->create($moduleName);
         } catch (Exception $e) {
             // This is a system, non-themed exception
         }
    }
}
```

and a module looks like:

```php
class Feed implements IModule
{
          public function __construct(App $app, OStatus $oStatus) 
          {
               // the constructor is called per $this->dependencyManager->create() and all dependencies are automatically "here"
               [...]
          }

         public function run()
         {
                try
                {
                   $app = $this->app;
                   $oStatus = $this->oStatus;
                } catch (HTTPException $e) {
                     // this is a themed, well defined exception
               }  
         } 
}
```

Reply:
It's called https://en.wikipedia.org/wiki/Inversion_of_control

Reply:
I like the idea of having an `App`-like class with a method that's the single call in the app entry points. Thats's what `Friendica\Core\Console` is about  (and what `bin/worker.php` and `bin/daemon.php` aren't) and I love this kind of simplicity.

In your example, it feels to me that the Router would de facto take that role with `getModule()` as the new `runFrontend()`, and I'd be fine with that.

Reply:
hm .. Console would be more like

console.php
```php
$dependencyManager = new DependencyManager(Dice $dependencyDILibrary);

$console = $dependencyManager->create($cmdName);

// in fact, this is exactly what console is currently doing like I thought about module
$console->execute();
```


```php
class AutomaticInstallationConsole extends Console
{
        public function __construct(App $app)
        {
        }
}
```

Because we now have an issue, where we must not load App, but with the current structure, we cannot skip this step and load everything.

The difference is that `getModule()` doesn't need any dependency except the route to the module. The current `App->runFrontend()` needs a fully compiled dependency structure (see the current constructor).

Reply:
Yep, sounds about right.